File: 001_two_sum/brute_force.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        """
        too slow...
        Complexity: O(n^2)
        Space: O(n^2) or O(1) if just use double loops
        """
        from itertools import combinations
        combs = combinations(range(len(nums)), 2)
        for comb in combs:
            if nums[comb[0]] + nums[comb[1]] == target:
                return [comb[0], comb[1]]
            

File: 189_rotate_array/brute_force.cpp
class Solution {
public:
 /*
  * Complexity: O(k*N)
  * Space: O(1)
  */
    void rotate(vector<int>& nums, int k) {
        for(int step = 0; step < k; ++step){
            int tmp_element = nums.back();
            int last_element;
            for(int i = 0; i<nums.size(); ++i){
                last_element = tmp_element;
                tmp_element = nums[i];
                nums[i] = last_element;
            }   
        }
    }
};
